ompi/mpi/java/java/Request.java

```diff
@@ -113,6 +117,22 @@ public final void cancel() throws MPIException
 	private native void cancel(long request) throws MPIException;
 
 	/**
+	 * Adds a buffer to the buffer array list.  This method should
+	 * be called by the internal api whenever a persistent request
+	 * is created and any time a request object, that has an associated
+	 * buffer, is returned from a non-blocking operation to protect 
+	 * the buffers from being garbage collected prematurely.
+	 * @param buf buffer to add to the array list
+	 */
+	protected final void addBufRef(Buffer buf)
+	{
+		if(this.buffers == null)
```

Comment:
Oh, I have a vague recollection of this now.

So -- the 8-space tabs is common in the Java world?

I'd still prefer the `{}` around all blocks, though; that's just defensive programming.